api/geth_backend.go

```diff
@@ -1127,10 +1137,13 @@ func (b *GethStatusBackend) ConvertToKeycardAccount(account multiaccounts.Accoun
 		for _, acc := range keypair.Accounts {
 			kc.AccountsAddresses = append(kc.AccountsAddresses, acc.Address)
 		}
-		err = accountDB.SaveOrUpdateKeycard(kc, uint64(time.Now().Unix()), true)
+		err = messenger.SaveOrUpdateKeycard(context.Background(), &kc)
 		if err != nil {
 			return err
 		}
+		// After `ConvertToKeycardAccount` function call the next step is logout, we need to give some time to the app (cause the
+		// password will be changed below) to dispatch the sync keypair message to paired devices, sent by `SaveOrUpdateKeycard` call.
+		time.Sleep(1 * time.Second)
```

Comment:
@IvanBelyakoff I don't like that either, but the reason is stated in the comments in those 2 places.

If you have a better idea of how to do that let me know?

Reply:
`ChangeDatabasePassword` does the logout, so no other way.

Reply:
Ahh I've thought you got it.

Reply:
@IvanBelyakoff so... when we're converting a profile keypair to:
- Start using a Keycard or
- Stop using a Keycard

sync keypair message is emitted, it should be received by all paired devices. That way paired devices know that they need to align with the change made on another device.

Now since synced messages are dispatched every 300ms without that "sleep" `ChangeDatabasePassword` does the `Logout` and synced messages for notifying paired devices about profile keypair migration to/from keycard are never emitted.

That's the point of that sleep, if you have a better idea I am eager to hear.

Reply:
You can check this function and see, https://github.com/status-im/status-go/blob/c3d2ff072319d12e7739e94854813ac7d75e4921/protocol/common/message_sender.go#L353

Reply:
@rasom @qfrank do you know if there is a better way to achieve the same (to be sure that the sync message is sent before `Logout` call) without using `sleep`?

Reply:
Could we leverage go coroutine and a wait here?

Reply:
Not sure what could be an event to trigger the end of go routine.

This function from vacp2p is called when we add a message to the store to be dispatched:
https://github.com/vacp2p/mvds/blob/d6b27d3843ffae66806cc969596af342c6e7a839/node/node.go#L271

This function from vacp2p is called when a message is dispatched for real:
https://github.com/vacp2p/mvds/blob/d6b27d3843ffae66806cc969596af342c6e7a839/node/node.go#L324

So all we can have is the message ID, but that's in the moment when we add it to the store, we don't know the moment when it's dispatched for real. The trick is that we do `Logout` from `ChangeDatabasePassword` call before the sync message gets dispatched and the paired device never receives it.

Reply:
waiting is not an option, as the same condition would happen if the user is offline for example, but messages should be retried even after login (at least we do for 1-to-1, and we should be doing the same pairing). This is assuming the message is scheduled at the datasync layer.

@saledjenic mentioned that even after login the message is not resent, so maybe there's some bug or timing issue, that's where we should have a look, but I think we can avoid the time.sleep